### ISSUE TITLE
PHPMD is returning "Avoid unused local variables such as '$var'." when using "named parameters" on PHP 8.

### ISSUE THREAD

**Issue by lucassabreu** [phpmd/phpmd#893]
- PHPMD version: 2.10.1
- PHP Version: 8.0.6
- Installation type: composer
- Operating System / Distribution & Version: Debian GNU/Linux 10 (buster) (tecnically is the docker image php:8.0)

### Current Behavior
PHPMD is returning "Avoid unused local variables such as '$token'." when using "named parameters" on PHP 8.

### Expected Behavior
The variable is being used as a parameter on `openssl_decrypt`, using named parameters or not. This error shouldn't be reported.

### Steps To Reproduce:

When phpmd runs on this code:

```php
<?php

namespace App;

class Hasher
{
    private const SEPARATOR = '::';

    public function decode(string $hash, string $method, string $secret): string
    {
        [$token, $iv] = explode(self::SEPARATOR, $hash);
        $iv = hex2bin($iv);

        $value = openssl_decrypt(
            data: $token,
            cipher_algo: $method,
            passphrase: $secret,
            iv: $iv,
        );

        return $value ?: '';
    }
}
```

It reports that the `$token` variable is not being used.

> Avoid unused local variables such as '$token'.

If I change to not use "named parameters" it stops reporting it.

```php
<?php

$value = openssl_decrypt(
    $token,
    cipher_algo: $method,
    passphrase: $secret,
    iv: $iv,
);
```

## Checks before submitting
* [x] Be sure that there isn't already an issue about this. See: [Issues list](https://github.com/phpmd/phpmd/issues)
* [x] Be sure that there isn't already a pull request about this. See: [Pull requests](https://github.com/phpmd/phpmd/pulls) (maybe #877?)
* [x] I have added every step to reproduce the bug.
* [x] If possible I added relevant code examples.
* [x] This issue is about 1 bug and nothing more.
* [x] The issue has a descriptive title. For example:  "JSON rendering failed on Windows for filenames with space".

**Comment by kylekatarnls** [comment]
Hello 👋 

Thanks for the report, indeed the way I handled the value of named argument prevents from discovering variable or any element inside. I made a fix to use the generic AST instead.

You can check if it solves your issue by running:
```
composer require pdepend/pdepend:dev-fix/named-argument-value-as-child
```

**Comment by lucassabreu** [comment]
hi @kylekatarnls , using your version fixed the problem, thanks.
